python/ray/autoscaler/_private/local/node_provider.py

```diff
@@ -68,6 +70,13 @@ def __init__(self, lock_path, save_path, provider_config):
                     if worker_ip not in list_of_node_ips:
                         del workers[worker_ip]
 
+                # Set external head ip, if provided by user.
+                # Useful if calling `ray up` from outside the network.
+                external_head_ip = provider_config.get("external_head_ip")
+                if external_head_ip:
```

Comment:
This is still puzzling for me

Reply:
Glad to remove it, but I will no longer be able to test.

Reply:
This works exactly like external and internal ips for aws.

Reply:
(In fact that's what it is.)

Reply:
I'd refer to the AWS implementation for more context.

Reply:
Added a more detailed docstring and removed from example-minimal.yaml, as this is primarily a debugging tool.

Reply:
Elaborating a little on our setup -- this isn't something I was aware of previously:
The autoscaler tells its node updaters to use internal node ip:
https://github.com/ray-project/ray/blob/226754d92c14a321b5eb7efbb174817d6df1bf56/python/ray/autoscaler/_private/autoscaler.py#L623
The local cluster launcher does get the `use_internal_ip` arg and so uses an external_ip:
https://github.com/ray-project/ray/blob/226754d92c14a321b5eb7efbb174817d6df1bf56/python/ray/autoscaler/_private/autoscaler.py#L623

Whether or not to use an external or internal ip affects the ssh command here: 
https://github.com/ray-project/ray/blob/226754d92c14a321b5eb7efbb174817d6df1bf56/python/ray/autoscaler/_private/command_runner.py#L347

To be able to `ray up` from my laptop, I had to use an external ip.

Reply:
Thought about it -- using public ips everywhere in this context wouldn't work, as ray internals infer internal ips but autoscaler would try use the public ips, leading to key errors in ResourceDemandScheduler...